docs/src/index.md

```diff
@@ -1,12 +1,173 @@
+If using the **colimit-of-representables** approach, you only need to specify relevant objects and morphism parts. The `K` part is empty because we want all the parts specified in `L` to be rewritten. You can use `homomorphisms` to automatically define the maps `l` and `r`.
+
+```julia
+yTeam = yoneda(Team)
+L = @acset_colim yTeam begin
+      (p1, p2)::Player
+      (team1, team2)::Team
+      IsMemberOf(p1) == team1
+      IsMemberOf(p2) == team2
+    end
+K = @acset_colim yTeam begin end
+R = @acset_colim yTeam begin
+      (p1, p2)::Player
+      (team1, team2)::Team
+      IsMemberOf(p1) == team2
+      IsMemberOf(p2) == team1
+    end
+l = only(homomorphisms(K, L))
+r = only(homomorphisms(K, R))
+```
+
+### 4. Construct the rule
+Use the `AlgebraicRewriting.Rule` constructor to create the rule. This assumes that a double-pushout (DPO) rewrite rule is being constructed. You may also construct an single-pushout (SPO), sesqui-pushout (SqPO), or pullback-pushout (PBPO) rule.
+
+```julia
+rule = Rule{:DPO}(l, r)
+```
+
+## Apply the rule
+### 5. Define the initial state. 
+Similarly, you can choose to define the acset using the static approach or the colimit-of-representable approach.
+
+- If using the **static approach**, you must fully specify the ACSet for the initial state.
+
+```julia
+state = @acset TeamStraightforward begin 
+    Player = 10  
+    Team = 2
+    IsMemberOf = [1, 1, 1, 1, 1, 2, 2, 2, 2, 2]
+    
+    TeamName = ["Home", "Away"]
+    HasName = [1, 2]
+end
+```
+
+- If using the **colimit-of-representable approach**, you only need to specify relevant objects and morphism parts.
+
+```julia
+state = @acset_colim yTeam begin
+  (p1, p2, p3, p4, p5, p6, p7, p8, p9, p10)::Player
+  (team1, team2)::Team
+  IsMemberOf(p1) == team1
+  IsMemberOf(p2) == team1
+  IsMemberOf(p3) == team1
+  IsMemberOf(p4) == team1
+  IsMemberOf(p5) == team1
+  IsMemberOf(p6) == team2
+  IsMemberOf(p7) == team2
+  IsMemberOf(p8) == team2
+  IsMemberOf(p9) == team2
+  IsMemberOf(p10) == team2
+end
+```
+
+### 6. Identify the match from the rule to the state
+This can be done manually or automatically. 
+
+- To **manually** identify the match, fully-specify an ACSet transformation. For this example, we would like to rule to swap `p5::Player` and `p6::Player`
+
+```julia
+match = ACSetTransformation(L, state, Player=[5, 6], Team=[1, 2], TeamName=[1, 2])
+```
+
+- To **automatically** identify the match, use the backtracking search algorithm provided by AlgebraicRewriting. This may returm multiple matches, so you can provide logic for deciding which match to select. 
+
+```julia
+matches = get_matches(rule, state)
+# insert logic to decide best match
+```
+
+### 7. Apply the rewrite rule 
+This executes the rewrite process using using the defined rule and match.
+
+```julia
+result = rewrite_match(rule, match)
+```
+
+This documentation provides a basic guide to using the AlgebraicRewriting package in Julia. 
```

Comment:
Does this line belong at the beginning?

Reply:
Yeah, you're right. It was a straggler. Moved it up.